pkg/kubelet/images/image_manager.go

```diff
@@ -73,19 +73,31 @@ func NewImageManager(recorder record.EventRecorder, imageService kubecontainer.I
 	}
 }
 
-// shouldPullImage returns whether we should pull an image according to
-// the presence and pull policy of the image.
-func shouldPullImage(pullPolicy v1.PullPolicy, imagePresent bool) bool {
-	if pullPolicy == v1.PullNever {
-		return false
+// imagePullPrecheck inspects the pull policy and checks for image presence accordingly,
+// returning (imageRef, err).
+func (m *imageManager) imagePullPrecheck(ctx context.Context, pullPolicy v1.PullPolicy, spec kubecontainer.ImageSpec) (string, error) {
+	// avoid fetching image if always pulling
+	if pullPolicy == v1.PullAlways {
```

Comment:
Using `switch pullPolicy` would be a little bit more readable in my opinion.

Reply:
Done